utils/generate_types/test/test.ts

```diff
@@ -323,6 +323,13 @@ playwright.chromium.launch().then(async browser => {
   console.log(await resultHandle.jsonValue());
   await resultHandle.dispose();
 
+  // evaluteHandle with two different return types
+  {
+    const handle = await page.evaluateHandle(() => '' as string | number);
+    const result = await handle.evaluate(value => value);
```

Comment:
evaluate was not passing TS before since handle was 'JSHandle<boolean> |JSHandle<number>' instead of 'JSHandle<boolean | number>'.